libraries/ESP8266mDNS/src/LEAmDNS2_Legacy.cpp

```diff
@@ -0,0 +1,1531 @@
+                   && (pService->setProbeResultCallback([this, p_hService, p_fnCallback](clsLEAMDNSHost::clsService& /*p_rMDNSService*/,
+                           const char* p_pcInstanceName,
+                           bool p_bProbeResult)->void
+        {
+            if (p_fnCallback)	// void((clsLEAMDNSHost_Legacy* p_pMDNSResponder, const char* p_pcServiceName, const hMDNSService p_hMDNSService, bool p_bProbeResult)
+            {
+                p_fnCallback(this, p_pcInstanceName, p_hService, p_bProbeResult);
+            }
+        })));
+    }
+    return bResult;
+}
+
+
+/*
+
+    PROCESS
+
+*/
+
+/*
+    clsLEAMDNSHost_Legacy::notifyAPChange
+
+*/
+bool clsLEAMDNSHost_Legacy::notifyAPChange(void)
+{
+    bool	bResult = true;
+
+    for (stcHostInformation::list::iterator it = m_HostInformations.begin(); ((bResult) && (it != m_HostInformations.end())); ++it)
+    {
+        bResult = (*it).m_pHost->restart();
+    }
+    return bResult;
+}
+
+/*
+    clsLEAMDNSHost_Legacy::update
+
+*/
+bool clsLEAMDNSHost_Legacy::update(void)
+{
+    bool	bResult = true;
+
+    for (stcHostInformation::list::iterator it = m_HostInformations.begin(); ((bResult) && (it != m_HostInformations.end())); ++it)
+    {
+        bResult = (*it).m_pHost->update();
+    }
+    return bResult;
+}
+
+/*
+    clsLEAMDNSHost_Legacy::announce
+
+*/
+bool clsLEAMDNSHost_Legacy::announce(void)
+{
+    bool	bResult = true;
+
+    for (stcHostInformation::list::iterator it = m_HostInformations.begin(); ((bResult) && (it != m_HostInformations.end())); ++it)
+    {
+        bResult = (*it).m_pHost->announce(true, true);
+    }
+    return bResult;
+}
+
+/*
+    clsLEAMDNSHost_Legacy::enableArduino
+
+*/
+clsLEAMDNSHost_Legacy::hMDNSService clsLEAMDNSHost_Legacy::enableArduino(uint16_t p_u16Port,
+        bool p_bAuthUpload /*= false*/)
+{
+    hMDNSService    hService = addService(0, "arduino", "tcp", p_u16Port);
+    if (hService)
+    {
+        if ((!addServiceTxt(hService, "tcp_check", "no"))
+                || (!addServiceTxt(hService, "ssh_upload", "no"))
+                || (!addServiceTxt(hService, "board", STRINGIZE_VALUE_OF(ARDUINO_BOARD)))
+                || (!addServiceTxt(hService, "auth_upload", (p_bAuthUpload) ? "yes" : "no")))
+        {
+            removeService(hService);
+            hService = 0;
+        }
+    }
+    return hService;
+}
+
+/*
+    clsLEAMDNSHost_Legacy::indexDomain
+
+*/
+bool clsLEAMDNSHost_Legacy::indexDomain(char*& p_rpcDomain,
+                                        const char* p_pcDivider /*= "-"*/,
+                                        const char* p_pcDefaultDomain /*= 0*/)
+{
+    bool	bResult = false;
+
+    const char*	cpcDomainName = clsLEAMDNSHost::indexDomainName(p_rpcDomain, p_pcDivider, p_pcDefaultDomain);
+    delete[] p_rpcDomain;
+    p_rpcDomain = 0;
+    if ((cpcDomainName)
+            && ((p_rpcDomain = new char[strlen(cpcDomainName) + 1])))
+    {
+        strcpy(p_rpcDomain, cpcDomainName);
```

Comment:
This is quite a trivial case where it will evidently not fail.
There are better replacements than strncpy(), like strdup(), or diving into the code to make it better.
```
    if ((cpcDomainName) && ((p_rpcDomain = new char[strlen(cpcDomainName) + 1])))
    {
        strcpy(p_rpcDomain, cpcDomainName);
        bResult = true;
    }
```